### ISSUE TITLE
s16b: [goldnaxe1] game locks in stage 2

### ISSUE THREAD

**Issue by jotego** [jotego/jtcores#294]
Transferred from [old repo](https://github.com/jotego/jts16/issues/5)

If the enemy is pushed off the cliff, the game doesn't let you advance. The go sign never shows up. Slashing the enemy lets you advance. Either killing method is ok in MAME.

![imagen](https://github.com/jotego/jtcores/assets/1863036/4af00698-7ea1-4de4-9ac8-af22f6bb67f8)

Music gets corrupted too.

First seen in https://github.com/jotego/jts16/commit/7dc5bde21cf062e2cf0722937b0d7ad39650e404

Fixed in https://github.com/jotego/jts16/commit/255a884c91ad0fe893b912ea3805225b567e6835 apparently related to interrupts, but appeared again after consolidating the repositories into jtcores. It happens in b29136b and before even though the interrupt generation is the same as in the old commit that fixed it.

It might be related to #172

Reported in **goldnaxe1** and **goldnaxeu**

**Comment by jotego** [comment]
- limiting the number of acknowledges interrupts to one per frame does not fix it

```
// interrupt generation
reg        last_vint, vint_n;

assign cpu_vpan = inta_n;
assign cpu_ipln = cpu_sel ? { vint_n, 2'b11 } : mmr[4][2:0];

always @(posedge clk) begin
    if( vint && !last_vint ) vint_n <= 0;
    if( !vint || !inta_n   ) vint_n <= 1;
end
```

**Comment by jotego** [comment]
The comparator/timer cs signals were not cleared. Clearing them fixes the problem:

```
    cmp_cs    <= 0;
    cmp2_cs   <= 0;
```

Fixed in 5c2ad075
